vpr/src/route/route_timing.cpp

```diff
@@ -333,6 +390,10 @@ bool try_timing_driven_route(const t_router_opts& router_opts,
         device_ctx.rr_switch_inf,
         route_ctx.rr_node_route_inf);
 
+    // Make sure template type ConnectionRouter is a ConnectionRouterInterface.
+    ConnectionRouterInterface* router_interface = &router;
+    (void)router_interface;
```

Comment:
Looks like this is unused, and can be removed?

Reply:
This is effectively "static_assert(is_router drived from ConnectionRouterInterface)"

Reply:
Could we do something more explicit like:
```
static_assert(std::is_base_of<ConnectionRouterInterface, ConnectionRouter>::value, "ConnectionRouter must implement the ConnectionRouterInterface");
```

Reply:
Yep, that works.  For some reason I thought it was only available in later C++'s